helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -245,6 +253,11 @@ public ZKHelixManager(String clusterName, String instanceName, InstanceType inst
 
     _stateListener = stateListener;
 
+    if (helixParticipantProperty != null) {
+      _helixParticipantProperty = helixParticipantProperty;
+    } else {
+      _helixParticipantProperty = HelixPropertyFactory.getInstance().getHelixParticipantProperty(zkAddress, clusterName);
+    }
```

Comment:
This could be refactored once we get rid of null's.

Reply:
already modified.